src/panels/lovelace/editor/hui-dialog-edit-card.ts

```diff
@@ -88,13 +107,40 @@ export class HuiDialogEditCard extends LitElement {
     this._dialog.close();
   }
 
+  private _toggleEditor() {
+    if (!this._elementConfig) {
+      return;
+    }
+    this._showUIEditor = !this._showUIEditor;
+    // This will center the dialog with the updated config
+    fireEvent(this._dialog, "iron-resize");
+  }
+
   private async _loadConfig() {
     this._cardConfig = await getCardConfig(this.hass!, this._cardId!);
+    this._loadElementConfig();
     await this.updateComplete;
     // This will center the dialog with the updated config
     fireEvent(this._dialog, "iron-resize");
   }
 
+  private async _loadElementConfig() {
+    const conf = yaml.safeLoad(this._cardConfig);
+    const elClass = customElements.get(`hui-${conf.type}-card`);
+    const element = document.createElement(`hui-${conf.type}-card`);
+
+    try {
+      console.log(this.hass);
+      console.log(conf);
+
+      this._elementConfig = (element as any).getElementConfig(conf, this.hass);
```

Comment:
You know what's so cool, once you have extracted the config element to it's own file and that declares it's type in the HTML Map, you won't need this cast as `any`